spmd/checkpoint/traverse.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+
+from typing import (
+    Callable,
+    Collection,
+    List,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Tuple,
+    TypeVar,
+    Union,
+    cast,
+)
+from torch.distributed._shard.checkpoint.metadata import (
+    STATE_DICT_TYPE,
+)
+from torch.distributed._shard.sharded_tensor.api import ShardedTensor
+from spmd import DTensor as DT
```

Comment:
nit: `from spmd.tensor import DTensor as DT`, we don't plan to offer DTensor as public in spmd package directly